### ISSUE TITLE
Analyze-Mediapackage WOH generates colons with config "set-publication-variables=true"

### ISSUE THREAD

**Issue by marwyg** [opencast/opencast#6009]
### TL;DR
- whisper generates a tag with a `:` in it (`lang:de`)
- analyze-mediapackage creates a variable that looks like this: `captions_prepared_hastag_lang:de`
- You get an parse error, when you try to use the variable from the analyze-mediapackage operation in an if-statement. Example:  if="NOT {captions_prepared_hastag_lang:de}"
- Should the tag of the whisper operation be changed?
- Should the analyze-mediapackage operation replace `:` with `_`?
- Something different?

### Info for DEV Meeting:

> Can you bring this up in the next dev meeting? There are probably other opinions regarding replacing chars and how compatible we need to be.

### Describe the bug

When setting the config `set-tag-variables` of the `analyze-mediapackage` WOH, the operation will generate the variable `captions_prepared_hastag_lang:de` for captions generated via whisper. (notice: it's `hastag` not `hashtag`). Whisper generates it automatically. I tried to use this for an if-statement, but I am getting a parse error, probably because of the colon `:`. I tried to escape it, which didn't work.

**To Reproduce**

1. Generate captions with whisper or alternatively just set a tag with `:` somewhere (when opencast lets you do this)
2. Use the `analyze-mediapackage` operation and set `set-tag-variables` to true
3. Create an if-statement within the workflow that uses the variable with the `:` in it
4. Run the workflow
5. Observe the workflow details and search for the new variable that was generated from your tag including the `:`
6. Workflow should fail with an parsing error

### Expected behavior

I don't know, maybe we could clarify in the dev meeting, where the error is. Shouldn't there be a colon in the first place? So do we change the whisper workflow? Or hould the analyze-mediapackage workflow replace the colon with an underscore for example?

**Stack trace**

```
org.opencastproject.workflow.api.WorkflowOperationException: Unable to parse execution condition 'NOT ${captions_prepared_hastag_lang:en}'
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.start(WorkflowOperationWorker.java:197)
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.execute(WorkflowOperationWorker.java:117)
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.runWorkflowOperation(WorkflowServiceImpl.java:722)
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.process(WorkflowServiceImpl.java:1754)
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2115)
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2081)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.IllegalArgumentException: line 1:6 extraneous input '<EOF>' expecting {'(', 'NOT', NUMBER, BOOL, STRING}
	at org.opencastproject.workflow.conditionparser.WorkflowConditionInterpreter$1.syntaxError(WorkflowConditionInterpreter.java:149)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportUnwantedToken(DefaultErrorStrategy.java:377)
	at org.antlr.v4.runtime.DefaultErrorStrategy.sync(DefaultErrorStrategy.java:275)
	at org.opencastproject.workflow.conditionparser.antlr.WorkflowConditionParser.booleanValue(WorkflowConditionParser.java:270)
	at org.opencastproject.workflow.conditionparser.antlr.WorkflowConditionParser.booleanTerm(WorkflowConditionParser.java:194)
	at org.opencastproject.workflow.conditionparser.antlr.WorkflowConditionParser.booleanExpression(WorkflowConditionParser.java:132)
	at org.opencastproject.workflow.conditionparser.WorkflowConditionInterpreter.interpret(WorkflowConditionInterpreter.java:155)
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.start(WorkflowOperationWorker.java:193)
	... 9 more
```

**Comment by mtneug** [comment]
> Should the tag of the whisper operation be changed?

Users can define custom tags, so dealing with unexpected characters in tags outside of Whisper is necessary. Moreover, we already have integrations expecting this tag format. I wouldn't change it at this point.

> Should the `analyze-mediapackage` operation replace : with _?

IIRC, we already replace stuff in flavors, right? So, there is precedence. On the other hand, this could, in theory, lead to conflicts with other tags (e.g. `lang:de` and `lang_de`). The problem is that the colon is a special character for properties that separates property name and default (see [this regex](https://github.com/opencast/opencast/blob/4b2882acecba4174f575879d8dad54a372ee05b0/modules/workflow-condition-parser/src/main/java/org/opencastproject/workflow/conditionparser/WorkflowConditionInterpreter.java#L44)).

A third option would be to allow any character (maybe except `{` and `}`) and change how default values are set.
